pkg/kubelet/cm/cgroup_manager_linux.go

```diff
@@ -560,85 +562,188 @@ func (m *cgroupManagerImpl) MemoryUsage(name CgroupName) (int64, error) {
 	return int64(val), err
 }
 
-// Get the memory limit in bytes applied to the cgroup
-func (m *cgroupManagerImpl) GetCgroupMemoryConfig(name CgroupName) (uint64, error) {
-	cgroupPaths := m.buildCgroupPaths(name)
-	cgroupMemoryPath, found := cgroupPaths["memory"]
-	if !found {
-		return 0, fmt.Errorf("failed to build memory cgroup fs path for cgroup %v", name)
-	}
-	memLimit, err := fscommon.GetCgroupParamUint(cgroupMemoryPath, "memory.limit_in_bytes")
-	if err != nil {
-		return 0, fmt.Errorf("failed to get memory.limit_in_bytes for cgroup %v: %v", name, err)
-	}
-	return memLimit, nil
+// Convert cgroup v1 cpu.shares value to cgroup v2 cpu.weight
+// https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2254-cgroup-v2#phase-1-convert-from-cgroups-v1-settings-to-v2
+func CpuSharesToCpuWeight(cpuShares uint64) uint64 {
+	return uint64((((cpuShares - 2) * 9999) / 262142) + 1)
 }
 
-// Get the cpu quota, cpu period, and cpu shares applied to the cgroup
-func (m *cgroupManagerImpl) GetCgroupCpuConfig(name CgroupName) (int64, uint64, uint64, error) {
-	cgroupPaths := m.buildCgroupPaths(name)
-	cgroupCpuPath, found := cgroupPaths["cpu"]
-	if !found {
-		return 0, 0, 0, fmt.Errorf("failed to build CPU cgroup fs path for cgroup %v", name)
-	}
-	cpuQuotaStr, errQ := fscommon.GetCgroupParamString(cgroupCpuPath, "cpu.cfs_quota_us")
+// Convert cgroup v2 cpu.weight value to cgroup v1 cpu.shares
+// https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2254-cgroup-v2#phase-1-convert-from-cgroups-v1-settings-to-v2
+func CpuWeightToCpuShares(cpuWeight uint64) uint64 {
```

Comment:
We should add a test for this.

Reply:
Added tests. Please review https://github.com/kubernetes/kubernetes/pull/102884/commits/6a18b9e612cc45a121a3587649ed6edaa266393a

Reply:
The tests look fine. Thanks!

Reply:
resolving conversation.